src/v/utils/tests/expiring_promise_test.cc

```diff
@@ -5,31 +5,6 @@
 #include <seastar/core/shared_future.hh>
 #include <seastar/testing/thread_test_case.hh>
 
-// testing that shared_promise doesn't immediately become available after set
```

Comment:
If we only had the `Note:...` above about the unexpected behavior this change wouldn't be concerning because code would probably have been designed for this delayed behavior and not been sensitive to the delay not being present (ie immediate availability).

But that we have a test for the quirk existing, makes it seem like the change on the Seastar side could be problematic. Like does the existence of this test mean that we depend on the quirky behavior?

Reply:
Yeah, I thought the same thing... so I reached out to @bharathv because although Denis created this test initially he has also been editing relevant code here. He mentioned he doesn't "think" we rely on this behavior, so my first intention is to remove it and see if any ducktape tests fail.